From 8eb3b8b4e1efe8f514dcb8266fb3892e2d5bf0fb Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Fri, 11 Apr 2025 11:38:55 +0200
Subject: [PATCH] generic: add missing of.h header to ar8327

It seems new kernel linux version reorganized the header include and now
of.h needs to be explicitly included. This should have been done from
when the driver was introduced.

Add the missing of.h header to fix compilation error in later kernel
version.

Co-authored-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/18455
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 target/linux/generic/files/drivers/net/phy/ar8327.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/target/linux/generic/files/drivers/net/phy/ar8327.c b/target/linux/generic/files/drivers/net/phy/ar8327.c
index 8da430fc92..503c801128 100644
--- a/target/linux/generic/files/drivers/net/phy/ar8327.c
+++ b/target/linux/generic/files/drivers/net/phy/ar8327.c
@@ -23,6 +23,7 @@
 #include <linux/lockdep.h>
 #include <linux/ar8216_platform.h>
 #include <linux/workqueue.h>
+#include <linux/of.h>
 #include <linux/of_device.h>
 #include <linux/leds.h>
 #include <linux/mdio.h>
-- 
2.43.0

